Add setpoint field editing support for Thermostat 6 devices, add dzVents support

dwmw2 opened the pull request:
Allow editing of ValueUnit, ValueStep, ValueMin, and ValueMax fields for pTypeThermostat6 devices in the temperature device edit dialog, matching the functionality available for setpoint devices in the utility controller.

Also add a dzVents device adapter for it.

dwmw2 committed:
Add setpoint field editing support for Thermostat 6 devices

Allow editing of ValueUnit, ValueStep, ValueMin, and ValueMax fields
for pTypeThermostat6 devices in the temperature device edit dialog,
matching the functionality available for setpoint devices in the
utility controller.

dwmw2 commented:
As noted elsewhere, setting the ValueUnit with this dialog *does* work, but just like with the existing standalone setpoints, the ValueUnit seems to be ignored. With some guidance on how it's actually *supposed* to work, we can fix those separately.

gizmocuz commented:
The ValueUnit is presented in the bigtext/top right of the widget. Also if you change this to  (degree sign)F it will use it for internal calculation.

dwmw2 commented:
> The ValueUnit is presented in the bigtext/top right of the widget. Also if you change this to (degree sign)F it will use it for internal calculation.

AFAICT it *isn't* though, not even for standalone setpoint devices (and thus not for the combined thermostat, which were based on those). My commit in https://github.com/dwmw2/domoticz/commit/b4ef5bcb9883525a1b9bdb3a90eec9b9fc439a35 tries to change that, but I wasn't sure exactly how it's supposed to behave.

waltervl commented:
If you create a virtual setpoint device you can change the value unit, even to RPM or A (Ampere), so not temperature related.
<img width="1200" height="598" alt="Screenshot_20251204-150310 Firefox" src="https://github.com/user-attachments/assets/b6797d86-cc98-4053-8042-3d9ca7a964dd" />

Edit: so the value sent is interpreted for the user and scripts as the value to use. No other calculation done. That could be different though for C/F temperature conversion.  Not sure how that works.

dwmw2 committed:
Add dzVents adapter for Thermostat 6 devices

Supports all pTypeThermostat6 subtypes:
- Temp/Setpoint
- Temp/Hum/Setpoint
- Temp/Baro/Setpoint
- Temp/Hum/Baro/Setpoint

Provides device.setPoint property and methods:
- updateSetPoint(setPoint) - updates only setpoint
- updateThermostat(temp, setpoint, hum, humStatus, baro, forecast) - updates all fields
  Fields are optional; nil values become empty strings in sValue

gizmocuz commented:
Thank you for this PR!!
